### PR TITLE
docs.rs metadata & license linter & fixes

### PR DESCRIPTION
**Pull request by rcoh**
## Motivation and Context
- docsrs builds should include all features
- docsrs builds should be restricted to a single target to reduce the docsrs build load
- all crates needs licenses / consistent authorship

## Description
- Lint / fix step for doc.rs metadata
- Follow up change is required to add `#![cfg_attr(docsrs, feature(doc_cfg_auto))]` to all crates
- Add 2 missing license files

## Testing
- [x] UTs
- [x] Ran fix/check on the repo 

## Checklist

- [x] I have updated `CHANGELOG.md` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `aws/SDK_CHANGELOG.md` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/sdk-lints/src/main.rs [hunk 1]
```diff
@@ -1,10 +1,30 @@
+use crate::lint_cargo_toml::{check_crate_author, check_crate_license, check_docs_rs, fix_docs_rs};
```
Comment (jdisanti) [inline tools/sdk-lints/src/main.rs:1]:
```suggestion
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0.
 */

use crate::lint_cargo_toml::{check_crate_author, check_crate_license, check_docs_rs, fix_docs_rs};
```

**Review by jdisanti — APPROVED** [review]
Nice!

File: aws/rust-runtime/aws-config/Cargo.toml [hunk 2]
```diff
@@ -69,3 +69,9 @@ serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 
 aws-smithy-client = { path = "../../sdk/build/aws-sdk/sdk/aws-smithy-client", features = ["test-util"] }
+
+[package.metadata.docs.rs]
+all-features = true
+targets = ["x86_64-unknown-linux-gnu"]
+rustdoc-args = ["--cfg", "docsrs"]
+# End of doc.rs metadata
```
Comment (david-perez) [inline aws/rust-runtime/aws-config/Cargo.toml:77]:
```suggestion
# End of docs.rs metadata
```

The typo exists in all the other `Cargo.toml`s.

File: tools/sdk-lints/src/lint_cargo_toml.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            package.authors
+        )
+    }
+    Ok(())
+}
+
+/// Check docsrs for a Cargo.toml
+///
+/// This function validates:
+/// - it is valid TOMl
```
Comment (david-perez) [inline tools/sdk-lints/src/lint_cargo_toml.rs:97]:
```suggestion
/// - it is valid TOML
```

File: tools/sdk-lints/src/lint_cargo_toml.rs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        bail!(
+            "missing `{}` in package author list ({:?})",
+            RUST_TEAM,
+            package.authors
+        )
+    }
+    Ok(())
+}
+
+/// Check docsrs for a Cargo.toml
```
Comment (david-perez) [inline tools/sdk-lints/src/lint_cargo_toml.rs:94]: Check `Cargo.toml` for docs.rs section ?

File: tools/sdk-lints/src/lint_cargo_toml.rs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+/// Set the default docs.rs anchor block
+///
+/// `[package.metadata.docs.rs]` is used as the head anchor. A comment `# End of docs.rs metadata` is used
+/// as the tail anchor.
+pub(crate) fn fix_docs_rs(path: impl AsRef<Path>) -> Result<bool> {
+    let mut cargo_toml = read_to_string(path.as_ref()).context("failed to read Cargo.toml")?;
+    let updated = replace_anchor(
+        &mut cargo_toml,
+        &("[package.metadata.docs.rs]", "# End of doc.rs metadata"),
```
Comment (david-perez) [inline tools/sdk-lints/src/lint_cargo_toml.rs:137]:
```suggestion
        &("[package.metadata.docs.rs]", "# End of docs.rs metadata"),
```

File: tools/sdk-lints/src/main.rs [hunk 6]
```diff
@@ -34,6 +69,18 @@ fn clap_app() -> App<'static, 'static> {
                         .required(false)
                         .long("readme"),
                 )
+                .arg(
+                    Arg::with_name("cargotoml")
```
Comment (david-perez) [inline tools/sdk-lints/src/main.rs:73]: Do you know about structopt? https://lib.rs/crates/structopt
I find it results in more readable CLI configurations.

File: tools/sdk-lints/src/lint_cargo_toml.rs [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    Ok(())
+}
+
+pub(crate) fn check_crate_author(path: impl AsRef<Path>) -> Result<()> {
+    let parsed = Manifest::from_path(path).context("failed to parse Cargo.toml")?;
+    let package = match parsed.package {
+        Some(package) => package,
+        None => bail!("missing `[package]` section"),
+    };
+    if SERVER_CRATES.contains(&package.name.as_str()) {
```
Comment (david-perez) [inline tools/sdk-lints/src/lint_cargo_toml.rs:81]: We'd also like to enjoy this linter script :(

**Review by david-perez — APPROVED** [review]

File: tools/sdk-lints/src/lint_cargo_toml.rs (hunk 7 above)
Comment (rcoh) [reply tools/sdk-lints/src/lint_cargo_toml.rs:81]: everything is currently applied to both except this author check, but that's a good point, I'll add the server owner. We'll just need to keep the list of `SERVER_CRATES` up to date

File: tools/sdk-lints/src/main.rs (hunk 6 above)
Comment (rcoh) [reply tools/sdk-lints/src/main.rs:73]: yeah all of our other CLIs use Clap directly so I was aiming for consistency. We can discuss as a team if we want to go one way or the other

**Pull request merged**